extract_test.go

```diff
@@ -59,6 +59,27 @@ func TestExtract(t *testing.T) {
 		}
 	})
 
+	t.Run("ValidateApp", func(t *testing.T) {
+		tests := []interface{}{
```

Comment:
add t.Parallel if these can run in parallel.

Reply:
Doesn't look like it can immediately, I get data races with `TestWithLogger` when trying to add that. 
Didn't look deeper into how this could be resolved. Let me know if you think this is a blocker.

Reply:
Dug into this. Looks like there's a legitimate race. TestWithLogger is marked as Parallel, but one of its tests modifies `os.Stderr`, which is what causes the data race. Fixing.

Reply:
nope, we can do this separately!